msal/confidential_client_application.go

```diff
@@ -45,52 +66,48 @@ func createInternalClientCredential(interfaceCred ClientCredentialProvider) (*ms
 	return msalbase.CreateClientCredentialFromAssertion(interfaceCred.GetAssertion().ClientAssertionJWT)
 }
 
-// SetHTTPManager allows users to use their own implementation of HTTPManager.
-func (cca *ConfidentialClientApplication) SetHTTPManager(httpManager HTTPManager) {
-	webRequestManager := createWebRequestManager(httpManager)
-	cca.clientApplication.webRequestManager = webRequestManager
-}
-
-// SetCacheAccessor allows users to use an implementation of CacheAccessor to handle cache persistence.
-func (cca *ConfidentialClientApplication) SetCacheAccessor(accessor CacheAccessor) {
-	cca.clientApplication.cacheAccessor = accessor
-}
-
 // CreateAuthCodeURL creates a URL used to acquire an authorization code. Users need to call CreateAuthorizationCodeURLParameters and pass it in.
-func (cca *ConfidentialClientApplication) CreateAuthCodeURL(authCodeURLParameters *AuthorizationCodeURLParameters) (string, error) {
+func (cca *ConfidentialClientApplication) CreateAuthCodeURL(authCodeURLParameters AuthorizationCodeURLParameters) (string, error) {
 	return cca.clientApplication.createAuthCodeURL(authCodeURLParameters)
 }
 
 // AcquireTokenSilent acquires a token from either the cache or using a refresh token
 // Users need to create an AcquireTokenSilentParameters instance and pass it in.
-func (cca *ConfidentialClientApplication) AcquireTokenSilent(
-	silentParameters *AcquireTokenSilentParameters) (AuthenticationResultProvider, error) {
+func (cca *ConfidentialClientApplication) AcquireTokenSilent(ctx context.Context, scopes []string, options *AcquireTokenSilentOptions) (*msalbase.AuthenticationResult, error) {
+	silentParameters := createAcquireTokenSilentParameters(scopes)
 	silentParameters.requestType = requests.RefreshTokenConfidential
 	silentParameters.clientCredential = cca.clientCredential
+	if options != nil {
+		silentParameters.account = options.Account
+	}
 	return cca.clientApplication.acquireTokenSilent(silentParameters)
 }
 
 // AcquireTokenByAuthCode is a request to acquire a security token from the authority, using an authorization code.
 // Users need to create an AcquireTokenAuthCodeParameters instance and pass it in.
-func (cca *ConfidentialClientApplication) AcquireTokenByAuthCode(
-	authCodeParams *AcquireTokenAuthCodeParameters) (AuthenticationResultProvider, error) {
+func (cca *ConfidentialClientApplication) AcquireTokenByAuthCode(ctx context.Context, scopes []string, redirectURI string, options *AcquireTokenByAuthCodeOptions) (*msalbase.AuthenticationResult, error) {
+	authCodeParams := createAcquireTokenAuthCodeParameters(scopes, redirectURI)
 	authCodeParams.requestType = requests.AuthCodeConfidential
 	authCodeParams.clientCredential = cca.clientCredential
+	if options != nil {
```

Comment:
Should cases like this be considered an error where we know that we need something off of the options struct?